src/libaktualizr/uptane/imagerepository.cc

```diff
@@ -264,6 +281,7 @@ void ImageRepository::updateMeta(INvStorage& storage, const IMetadataFetcher& fe
     // If we don't, attempt to fetch the latest.
     if (fetch_snapshot) {
       fetchSnapshot(storage, fetcher, local_version);
```

Comment:
I suppose we do have tests for the case when the new version of snapshot meta is invalid, and the same for targets role meta?

Reply:
`fetchSnapshot` is not affected by the current patch, since there are no changes inside `verifySnapshot` (it only logs errors if `prefetch == true`).
`fetchTargets` was affected, since `verifyTargets` also may log errors right after a download (like it does when loading a corrupted file locally). Verifying the error after a invalid downloaded file is not part of the current test. I'm looking at the best way to add that.

Reply:
I expected that the uptane test vector should include such tests since it's a common threat - new version meta are invalid.

Reply:
The uptane vectors test does not cover that exact scenario, where the targets file does not match the expected hash.  The closest it gets is a case where the targets has an invalid `_type` (but still a valid hash). Besides that, the test framework does not provide a way to specify expected log output for the vector in a generic way.
I've added a new test inside `test_imagerepo_failure.py`. The fixtures in use there were enough for me to generate the error sequence where the targets file has an invalid hash, and analyse the logs output, looking for the expected messages.